modules/standard/Regex.chpl

```diff
@@ -611,12 +623,65 @@ record regex {
 
   proc init=(x: regex(?)) {
     this.exprType = x.exprType;
-    this.home = x.home;
-    this._regex = x._regex;
-    this.complete();
-    on home {
-      qio_regex_retain(_regex);
+    /* always bring the regex local */
+    this.home = here;
+    /* if it's local, retain and avoid the recompile (thread safe) */
+    if (x.home == here) {
+      this._regex = x._regex;
+      this.complete();
+      qio_regex_retain(x._regex);
+    } else {
+      /* otherwise recompile locally */
+      this.complete();
+      var serialized = x._serialize();
+      this._deserialize(serialized);
```

Comment:
Remind me, does `qio_regex_create_compile` end up bumping the reference count in the runtime, so we don't need to do that here?

Reply:
Yep